Assets/Scripts/Player/PlayerController.cs

```diff
@@ -73,9 +71,14 @@ public class PlayerController : MonoBehaviour {
 			scale.x = -1;
 		transform.localScale = scale;
 
-		if (rb2D.velocity.x == 0)
+		if (Mathf.Abs(rb2D.velocity.x) < 0.01f)
 			velocity.x = 0;
 
+		if (Mathf.Abs(rb2D.velocity.y) < 0.01f && !isGrounded) {
+			doJump = false;
+			doAirJump = false;
+		}
+
```

Comment:
I'm a bit confused over what this is doing.
If it's making sure to cancel any potential jump if we walk of ground, then why are we checking the y velocity?

Reply:
This is what prevents the player from floating in the roof by canceling any potential jumps if the velocity becomes too low.

Reply:
Oh, i see that now. That makes sense!